functional-tests/test-bed/src/main/java/org/arquillian/smart/testing/ftest/testbed/rules/GitClone.java

```diff
@@ -2,21 +2,38 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.FileVisitOption;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
 import java.util.logging.Logger;
 import org.eclipse.jgit.api.Git;
 import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.Repository;
 import org.junit.rules.ExternalResource;
 
 public class GitClone extends ExternalResource {
 
-    private static final String ORIGIN = "https://github.com/arquillian/smart-testing-dogfood-repo.git";
-    private static final String REPO_NAME = ORIGIN.substring(ORIGIN.lastIndexOf('/') + 1).replace(".git", "");
+    private static final Logger LOGGER = Logger.getLogger(GitClone.class.getName());
 
+    private final String gitRepo;
     private String gitRepoFolder;
+    private File tempFolder;
 
-    public static final Logger LOGGER = Logger.getLogger(GitClone.class.getName());
+    public GitClone() {
+        try {
+            this.gitRepo = new URL("https://github.com/arquillian/smart-testing-dogfood-repo.git").toExternalForm();
```

Comment:
Because `new URL("https://github.com/arquillian/smart-testing-dogfood-repo.git")` throws an exception which would need to end up in constructor signature and make it awful?